esiclient/v1/switch.py

```diff
@@ -172,3 +171,284 @@ def take_action(self, parsed_args):
         print("Disabling access to {0}".format(switchport))
 
         ironic_client.node.vif_detach(port.node_uuid, np_uuid)
+
+
+class EnableTrunkPort(command.ShowOne):
+    """Configure Switchport Trunk Access"""
+
+    log = logging.getLogger(__name__ + ".EnableTrunkPort")
+
+    def get_parser(self, prog_name):
+        parser = super(EnableTrunkPort, self).get_parser(prog_name)
+        parser.add_argument(
+            "switch",
+            metavar="<switch>",
+            help=_("Switch"))
+        parser.add_argument(
+            "switchport",
+            metavar="<switchport>",
+            help=_("Switchport"))
+        parser.add_argument(
+            "vlan_id",
+            metavar="<vlan_id>",
+            help=_("VLAN ID"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        switch = parsed_args.switch
+        switchport = parsed_args.switchport
+        vlan_id = parsed_args.vlan_id
+
+        # get associated port and node
+        ironic_client = self.app.client_manager.baremetal
+        ports = ironic_client.port.list(detail=True)
```

Comment:
I think this code block is used a lot in different functions (line 207 - line 213). Should we create a function like utils.get_barmetal_port() to avoid duplication?